### ISSUE TITLE
Week of 10/29

### ISSUE THREAD

**Issue by ericaddison** [ericaddison/Hailing_Frequencies#4]
@mspagon @psigourney 
Hey guys,

Soooo... Now that we've all rested a day from Phase 3... Let's talk about our next steps. We have three things that will be due at the end of this week, namely:
- Video Demo of Android Miniproject
- Marketing Document for Final Project
- Design Document for Final Project

I think the easiest thing to do in terms of low ambiguity would be for each of us to be responsible for one of these items. We should all provide input for all items, but each would be "owned" by one person. I could envision the assigment being like:
- Mike: Demo video
- Patrick: Marketing document
- Eric: Design document

Clearly we would all need to collaborate on the presentations... Let me know if that sounds good, or if you guys would prefer a different responsibility, or a completely different approach :)

Thanks!
Eric

**Comment by psigourney** [comment]
Sounds fine.

**Comment by ericaddison** [comment]
Cool, thanks. I guess we will talk on Wednesday and see how we're all doing.

**Comment by mspagon** [comment]
Sounds good to me.

**Comment by psigourney** [comment]
From the mockups, it looks like we're going to have a chatroom app, where people can create their own rooms or join others' rooms, includes the ability to post images into the chat.  Named "Hailing Frequencies".  Anything else?  Any unique 'hook' or gimmick that would set us apart?  The marketing document says the app should serve an unmet need, so I was going to run with the whole AOL Instant Messenger being retired in December, and this being a replacement for AOL chatrooms.

**Comment by ericaddison** [comment]
I think three things set this apart from a normal chatroom app:
1) The conversations are meant to be finite duration, and for specific purposes (e.g. class discussion for a semester class)
2) Users can interact via chat-room interface (web or mobile), text, or email.
3) (perhaps the most important) Comm option is obfuscated. Users get to communicate without ever revealing any permanent information (phone #, email address, or even user name).

**Comment by psigourney** [comment]
But in the mockup for the sign-in page, it has email / google account fields?

**Comment by ericaddison** [comment]
Right... That will just be for sign up / registration. WE (the server) get to know their contact info so that our app can send them converstaion updates, but the users won't get to see each other's contact info.

Also we should get their info and do some sort of verification step like most sites do, so that you can't just sign up with someone else's email or phone #...

Probably you sign up with a primary email, but then you can add other emails and phone #s (each need verification). In the "future", it would also be good to integrate with other messaging apps (FB, slack, etc)

**Comment by psigourney** [comment]
Got it.

WIP document...

https://drive.google.com/open?id=11O8itkpZ0X8H0_GG-mSP_2TtK4CScWcv1yAR_Kbv7bA

**Comment by ericaddison** [comment]
Design document almost done (pending you guys' approval)

https://drive.google.com/open?id=1pGthnf41xkZmG5-pfZmG8qDTf_H-00PAOvjbajU8sk8

**Comment by mspagon** [comment]
Were we going to meet tonight? or just keep it to this thread?

Michael

**Comment by psigourney** [comment]
I had a question about the mobile component... is it an Android app using the internet or is it some sort of SMS text thing?

**Comment by mspagon** [comment]
I believe it's both... a mobile app for full functionality, but also a way partake in simple conversations via e-mail and SMS?

correct me if i'm wrong

**Comment by ericaddison** [comment]
Sooo ... I think one possibility is to have an Android app that uses the internet, so it would interact via a web API, and also have pure texting interaction.

And yeah @mspagon , I think we should meet tonight.

**Comment by ericaddison** [comment]
https://drive.google.com/open?id=1u-7paXWY4U8pIsjoZBo_iK3HXA16i-JhBs_d52F19xE

**Comment by ericaddison** [comment]
I'm thinking we might want to discuss the timeline and work distribution a little more, as in maybe we all are responsible for more separate pieces...

One possibility would be someone take web front end, someone take Android front end, someone take general server side (API and NDB), and we all take one of the comm options (Firebase, MailGun, Twilio). With something like that, maybe the parallelization of work would be better?

Lets chat about it tomorrow around class. :)

@mspagon @psigourney

**Comment by ericaddison** [comment]
@psigourney @mspagon 
Hey guys,

Mike, Patrick and I talked at the end of class about a better way to approach the project for timeline and milestones. I made a new timeline here:
https://docs.google.com/spreadsheets/d/1_USlIN1TR2E-l4ZHXYZ7DVFxl12J0hhvotoq0HBmmVM/edit#gid=0

Basically, we should be more parallel. For the first "phase", lets split the work like this:
- Patrick: Data model (python classes, NDB schema, etc)
- Eric: Anticipated Web API endpoints (management stuff like: get all conversations, join a conversation, and actual use stuff like post new message, edit message, etc)
- Mike: Admin/Test dashboard (list users, rebuild database, test functionality, etc)

Obviously we will need to be communitcating during this process :) 

The next steps, in a week and a half or so, will be:
- Android front end
- Web front end
- Comm APIs

We can talk about who does which of these steps later.

This way we can be making good progress on multiple fronts in parallel, and stay out of each other's way to some extent (working on different files), but also be evolving the API/data model at the same timeso they build off each other. The admin/test dashboard should exercise the functions written for both the model and the API, so we can also get a good sense of how our components work and what we will need. Then, when it comes to making the front ends, we just need to make pages/activities and connect to the different API services we've created (and tested).

Sound OK???
